src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -371,7 +395,71 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 			case ITermOscPt.SetMark: {
 				this._createOrGetCommandDetection(this._terminal).handleGenericCommand({ genericMarkProperties: { disableCommandStorage: true } });
 			}
+			default: {
+				// Checking for known `<key>=<value>` pairs.
+				const { key, value } = this._parseKeyValueAssignment(command);
+
+				if (value === undefined) {
+					// No '=' was found, so it's not a property assignment.
+					return true;
+				}
+
+				switch (key) {
+					case ITermOscPt.CurrentDir:
+						// Encountered: `OSC 1337 ; CurrentDir=<Cwd> ST`
+						this._updateCwd(value);
+						return true;
+				}
+			}
 		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	private _doHandleCmderSequence(data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command, ...args] = data.split(';');
+		switch (command) {
+			case CmderOscPt.Code9:
+				if (args.length) {
+					this._updateCwd(args[0]);
+				}
+				return true;
+		}
+
+		// Unrecognized sequence
+		return false;
+	}
+
+	/**
+	 * Some escape sequences are not well known/documented, so we handle them in
+	 * an ad-hoc/one-off manner. Also, note that in some cases the `ident` value
+	 * itself acts like a command (not an terminal/emulator identifier). So, it
+	 * seems logical to have a slack method that deals with such cases.
+	 */
+	private _doHandleUnknownSequence(ident: number, data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command] = data.split(';');
+		switch (ident) {
+			case ShellIntegrationOscPs.Unknown7: {
+				// Checking for: `OSC 7 ; scheme://cwd ST`
+				if (command.startsWith('scheme://')) {
+					// TODO: I'm not sure `scheme` here is literal or can be `file` or something else.
+					// TODO: Possibly the path is URL-encoded, but I have no means to test it.
+					const cwd = command.substring(9);
+					this._updateCwd(cwd);
+					return true;
+				}
```

Comment:
> Is the scheme:// a literal string or is a placeholder for actual scheme identifier? Naturally it should be the latter.
> Since the URI format, naturally the CWD value should be URI-encoded. Is it true here?

I haven't looked into this too deeply yet, here are some references to help:

- This seems very comprehensive: https://wezfurlong.org/wezterm/shell-integration.html#osc-7-escape-sequence-to-set-the-working-directory
- https://github.com/neovim/neovim/issues/2252
- https://gitlab.freedesktop.org/terminal-wg/specifications/-/issues/20
- https://iterm2.com/documentation-escape-codes.html (search "OSC 7")

Reply:
Thanks for the references. I also did some tests (will now comment the details) to make sure we're implementing it right.